Update dependencies

kelvinqian00 opened the pull request:
## Tasks
- [x] Updated xapi-schema dep to latest SHA
- [x] Update project-pan dep to latest SHA

TODO: Update pedestal deps to 0.5.9 (to match lrs)?

## Effects
- Profile validation error messages may change
- Fixed bug with attachment SHA2 validation
- Fixed bug with Statement duration validation
- Fixed bug with Statement email validation
- Statement UUID validation now accepts versions other than 4
- Statement IRIs now support non-ASCII chars

milt reviewed (approved):
This looks good, it reminds me we should migrate this to gh actions.

kelvinqian00 commented on src/main/com/yetanalytics/datasim/input.clj:
```diff
@@ -1,35 +1,44 @@
+                      :patterns  []}))]
+    (pat/get-edges (pat/create-graph combined-profile))))
+
+;; ::profiles is unused, but needs to stay or else tests will break
```
The only reason I'm not also deleting or modifying the tests is because I'm not sure if that would technically count as a breaking change.

kelvinqian00 replied:
Decided to switch over tests and delete unused specs anyways: https://github.com/yetanalytics/datasim/pull/65/commits/13512a381338024242b20faa6804f6ffcb1b2017. Doing so after helped me uncover the bug https://github.com/yetanalytics/project-pan/pull/31 in Pan.

kelvinqian00 commented on src/main/com/yetanalytics/datasim/util/errors.clj:
```diff
@@ -0,0 +1,18 @@
+(ns com.yetanalytics.datasim.util.errors
+  (:require [clojure.spec.alpha :as s]))
+
+(defn explain-to-map-coll
+  "Convert spec error data for personae, alignments, and parameters
+   into a map coll acceptable to the Datasim UI."
+  [spec-kw spec-ed]
+  (mapv
+   (fn [problem]
+     (let [epath (into [spec-kw] (:in problem))
+           estr  (-> problem
+                     (assoc ::s/problems [problem])
+                     s/explain-printer
+                     with-out-str)]
+       {:path epath
+        :text estr
+        :id   epath}))
+   (::s/problems spec-ed)))
```
This is what I have for each map - the `:path` and `:text` are self-explanatory, but for now I'm setting `:id` to be the path value. I'm not sure if it's worth investing time making a better ID value or if it should just be removed from the UI.

cliffcaseyyet replied:
that makes sense.

kelvinqian00 replied:
Ok I decided to update the `:id` value to be of the form `specname-idx` (e.g. `profiles-0`) in https://github.com/yetanalytics/datasim/pull/65/commits/a2720dd580f431474eb69797f09fb7a4d5ff66ab. It seems like the cleanest solution to me.

BTW how do I find the value of the ID in developer tools?

cliffcaseyyet commented on src/main/com/yetanalytics/datasim/input.clj:
```diff
@@ -56,6 +65,35 @@
+  (if (vector? profile-coll)
+    (let [prof-errs (pan/validate-profile-coll profile-coll
+                                               :syntax? true
+                                               :pattern-rels? true
```
just pattern, not template?

kelvinqian00 replied:
`:template-rels?` is about relationships between Templates and Concepts. `:pattern-rels?` is about relations between Patterns and Templates, as well as between different Patterns.

kelvinqian00 commented:
Decided to introduce additional changes since your approval:
1. Added an entry in the Changelog. I arbitrarily denoted the first version as 0.1.0, and this one as 0.1.1*; let me know if it should be something different.
2. Switched over tests to use the new validate functions instead of specs.

* I realized that I _did_ introduce a minor breaking change, in that single-profile validation will now fail if it has incorrect Pattern relations, whereas before it did not. I don't think this should be reverted since you would want your Pattern relations to be valid anyways, but let me know if it should be.

cliffcaseyyet reviewed (approved):
already approved but i guess it unapproved?

kelvinqian00 commented:
@cliffcaseyyet I re-requested reviews after making a bunch of changes. BTW I'm still waiting for you to approve https://github.com/yetanalytics/project-pan/pull/31 to get that tagged version of Pan up.
